app.js

```diff
@@ -274,6 +275,65 @@ const assistant = new Assistant({
 
 app.assistant(assistant);
 
+/**
+ * `app_mention` event allows your app to receive message events that directly
+ * mention your app. The app must be a member of the channel/conversation to
+ * receive the event. Messages in a DM with your app will not dispatch this event,
+ * event if the message mentions your app.
+ *
+ * @see {@link https://docs.slack.dev/reference/events/app_mention/}
+ */
+app.event('app_mention', async ({ event, client, logger, say }) => {
+  try {
+    const { channel, text, team, user } = event;
+    const thread_ts = event.thread_ts || event.ts;
```

Comment:
note: This line is very important. at-mentions in a public channel do not have a `thread_ts` so we use the `ts`. Without a value, the text streaming API will fail.